pwndbg/dbg/lldb/repl/__init__.py

```diff
@@ -679,60 +687,96 @@ def process_launch(driver: ProcessDriver, relay: EventRelay, args: List[str], db
-
     targets = dbg.debugger.GetNumTargets()
     assert targets < 2
     if targets == 0:
         print(message.error("error: no target, create one using the 'target create' command"))
         return
 
+    # TODO/FIXME: This should ask:
+    # 'There is a running process, detach from it and attach?: [Y/n]'
     if driver.has_process():
         print(message.error("error: a process is already being debugged"))
         return
 
-    # The first two arguments - executable name and wait_for_launch - don't
-    # matter, we set them later. The third one is required, as it tells LLDB the
-    # attach should be asynchronous.
-    params = lldb.SBAttachInfo(None, False, True)
-
-    if args.name is not None:
-        params.SetExecutable(args.name)
-    if args.pid is not None:
-        params.SetProcessID(args.pid)
-    params.SetWaitForLaunch(args.waitfor)
-    if getattr(args, "continue"):
-        params.SetResumeCount(1)
-    params.SetIgnoreExisting(not args.include_existing)
-
     io_driver = get_io_driver()
+
     result = driver.attach(
         dbg.debugger.GetTargetAtIndex(0),
         io_driver,
-        params,
+        info,
     )
 
     if not result.success:
         print(message.error(f"Could not attach to process: {result.description}"))
         return
 
     # Continue execution if the user has requested it.
-    if getattr(args, "continue"):
+    if cont:
         # Same logic applies here as in `process_launch`.
         relay._set_ignore_resumed(1)
-
         driver.cont()
     else:
         # Same logic applies here as in `process_launch`.
         dbg._trigger_event(EventType.STOP)
 
 
+def process_attach(driver: ProcessDriver, relay: EventRelay, args: List[str], dbg: LLDB) -> None:
+    """
+    Attaches to a process with the given arguments.
+    """
+    args = parse(args, process_attach_ap, process_attach_unsupported)
+    if not args:
+        return
+
+    # The first two arguments - executable name and wait_for_launch - don't
+    # matter, we set them later. The third one is required, as it tells LLDB the
+    # attach should be asynchronous.
+    info = lldb.SBAttachInfo(None, False, True)
+
+    if args.name is not None:
+        info.SetExecutable(args.name)
+    if args.pid is not None:
+        info.SetProcessID(args.pid)
+    info.SetWaitForLaunch(args.waitfor)
+
+    do_continue = getattr(args, "continue", False)
+    if do_continue:
+        info.SetResumeCount(1)
+    info.SetIgnoreExisting(not args.include_existing)
+
+    _attach_with_info(driver, relay, dbg, info, cont=do_continue)
+
+
+def attach(driver: ProcessDriver, relay: EventRelay, args: List[str], dbg: LLDB) -> None:
+    """
+    Attaches to a process with the given name or pid based on regex match.
+    Used for `_regexp-attach <pid|name>` (alias for `attach <pid|name>`)
+    Note: for some reason, `attach` does not really take a regex for process name.
+    """
+    assert len(args) == 1, "Expected 1 argument: <pid> or <name>"
```

Comment:
Are you sure an assertion here is the way to go? Doesn't look like we check for the number of arguments anywhere before this, and having `AssertError` be raised here will bring Pwndbg down. Why not do it the same way user errors are handled in the other commands?

```suggestion
    if len(args) != 1:
        print(message.error("Expected 1 argument: <pid> or <name>"))
        return
```